test/e2e/instrumentation/tracing/apiserver.go

```diff
@@ -0,0 +1,190 @@
+var _ = instrumentation.SIGDescribe("[Feature:APIServerTracing]", func() {
+	f := framework.NewDefaultFramework("apiserver-tracing")
+	var c clientset.Interface
+	var otelPod *v1.Pod
+
+	ginkgo.BeforeEach(func() {
+		config, err := framework.LoadConfig()
+		framework.ExpectNoError(err)
+		// Use otelhttp with the client-go client to ensure context is propagated.
+		config.Wrap(func(rt http.RoundTripper) http.RoundTripper {
+			return otelhttp.NewTransport(rt)
+		})
+		c, err = clientset.NewForConfig(config)
+		framework.ExpectNoError(err)
+
+		ginkgo.By("Creating an opentelemetry collector on the master node, which logs spans to stdout.")
+		masterNode, err := masterNodeName(c)
+		framework.ExpectNoError(err)
+		otelPod, err = c.CoreV1().Pods(f.Namespace.Name).Create(context.TODO(), opentelemetryCollectorPod(masterNode), metav1.CreateOptions{})
+		framework.ExpectNoError(err)
+
+		_, err = c.CoreV1().ConfigMaps(f.Namespace.Name).Create(context.TODO(), opentelemetryConfigmap(), metav1.CreateOptions{})
+		framework.ExpectNoError(err)
+
+		ready := e2epod.CheckPodsRunningReady(f.ClientSet, f.Namespace.Name, []string{podName}, 20*time.Second)
+		framework.ExpectEqual(ready, true)
+	})
+
+	ginkgo.It("should send a request with a sampled trace context, and observe child spans from the collector pod", func() {
+		e2eskipper.SkipUnlessSSHKeyPresent()
+
+		ginkgo.By("Setting up OpenTelemetry")
+		tp := sdktrace.NewTracerProvider()
+		otel.SetTracerProvider(tp)
+		otel.SetTextMapPropagator(traces.Propagators())
+
+		ginkgo.By("Creating a context with a  parent span")
+		ctx, span := tp.Tracer("apiservertest").Start(context.Background(), "OpenTelemetrySpan")
+
+		traceID := span.SpanContext().TraceID()
+		ginkgo.By(fmt.Sprintf("Checking for Trace ID: %v in logs.", traceID.String()))
+		gomega.Eventually(func() error {
+			// Send any request using the context with a sampled span
+			_, err := c.CoreV1().Nodes().List(ctx, metav1.ListOptions{})
+			if err != nil {
+				return err
+			}
+
+			// Get logs from the opentelemetry collector pod on the master.
+			// We must use SSH because we can't fetch logs from master pods
+			// using the pod logs subresource.
+			result, err := e2essh.SSH(
+				fmt.Sprintf("sudo cat /var/log/pods/%v_%v_%v/%v/*", f.Namespace.Name, podName, otelPod.UID, containerName),
+				framework.APIAddress()+":22",
+				framework.TestContext.Provider,
+			)
+			logs := result.Stdout
+			if err != nil {
+				return err
+			}
+			if result.Stderr != "" {
+				return fmt.Errorf("non-empty stderr when querying for logs on the master: %v", result.Stderr)
+			}
+			// Check the opentelemetry collector logs to see if they contain our trace ID
+			if strings.Contains(logs, traceID.String()) {
+				return nil
+			}
+			return fmt.Errorf("failed to find trace ID %v in log: \n%v", traceID.String(), logs)
+		}, 1*time.Minute, 10*time.Second).Should(gomega.BeNil())
+
+	})
+})
+
+func masterNodeName(c clientset.Interface) (string, error) {
```

Comment:
In an HA cluster, it will only find the first master node.  I'm hoping we can address this with the same mechanism as we do to remove the need for SSH.